### ISSUE TITLE
Armeria does not set a negativeTtl or allow a negativeTtl to be configured for the client

### ISSUE THREAD

**Issue by AngerM** [line/armeria#3938]
The Netty DNS resolvers do not cache negative responses by default, which can cause typos (or other dns related outages/mishaps) to make armeria overwhelm DNS infrastructure by flooding it with requests to which it gets a negative response.

negativeTtl is a config you can pass to the netty DnsNameResolverBuilder:
https://netty.io/4.1/api/io/netty/resolver/dns/DnsNameResolverBuilder.html#negativeTtl-int-

**Comment by ikhoon** [comment]
Did you mean that https://javadoc.io/static/com.linecorp.armeria/armeria-javadoc/1.13.2/com/linecorp/armeria/client/DnsResolverGroupBuilder.html#negativeTtl(int) did not work as you exacted?
FWIW, negative TTL is disabled by default and you can enable the option with:
```java
ClientFactory.builder()
             .domainNameResolverCustomizer(builder -> {
                 builder.negativeTtl(5 /* seconds */);
             });
```

**Comment by AngerM** [comment]
@ikhoon we use https://javadoc.io/doc/com.linecorp.armeria/armeria-javadoc/latest/com/linecorp/armeria/client/endpoint/dns/DnsAddressEndpointGroupBuilder.html
and it would be great to have negative TTL support on that

**Comment by AngerM** [comment]
Hmm yeah, was able to implement this, but the API is super unintuitive.
EX: this is what you have to do to enable a negative ttl
```
                Clients.builder(
                    "$SERIALIZATION_FORMAT+$SESSION_PROTOCOL",
                    DnsAddressEndpointGroup
                        .builder(hostname)
                        .port(port())
                        .ttl(MIN_TTL_SECONDS, MAX_TTL_SECONDS)
                        .build()
                ).factory(
                    ClientFactory.builder()
                        .domainNameResolverCustomizer { builder ->
                            builder.negativeTtl(MAX_TTL_SECONDS)
                        }
                        .build()
                )
```
as opposed to the normal TTL settings that live on the DnsAddressEndpointGroup

**Comment by minwoox** [comment]
Oops, we can't set negative TTL for `DnsAddressEndpointGroup`. 😅 
`domainNameResolverCustomizer` is for a `WebClient` not for a `DnsAddressEndpointGroup`.
We need to provide a setter for the builder or resolverConfigurator(`Consumer<DnsNameResolverBuilder>`).

**Comment by AngerM** [comment]
I will see if I have time to add it. We have some slowdown coming up with the holidays so hopefully I can take some time away to do it.

**Comment by trustin** [comment]
@ikhoon Was this resolved already in #4133 ?

**Comment by ikhoon** [comment]
Indeed! I forgot to close this issue. This issue was fixed by #4133.
